File: src/main/java/org/splittydupe/startup/model/Receipt.java
package org.splittydupe.startup.model;

import com.google.cloud.Timestamp;
import java.util.List;
import java.util.Map;

import lombok.Data;
import lombok.Builder;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Receipt {
    private String uid;

    private String vendor;

    private String receiptDate;

    @Builder.Default
    private String currency = "USD";

    private double subtotal;

    private double tax;

    private double taxPercentage;

    private double tip;

    private double tipPercentage;

    private double total;

    private List<LineItem> lineItems;

    // User ID - "anonymous" for non-logged-in users
    private String userId;

    private Timestamp createdAt;

    // Expiration for anonymous receipts (7 days from creation)
    private Timestamp expiresAt;

    @Builder.Default
    private String paymentStatus = "pending"; // "pending", "partial", "complete"

    // Maps assignee name to payment status ("paid" or "unpaid")
    private Map<String, String> assigneePaymentStatus;

    // "proportional" - distribute based on line item costs (default)
    // "even" - distribute evenly across all assignees
    @Builder.Default
    private String taxTipDistribution = "proportional";
}

File: react-app/src/types/receipt.ts
export interface LineItem {
    name?: string;
    price?: number;
    quantity?: number;
    assignees?: string[];
    numAssignees?: number;
    assigneesText?: string;
    splitMode?: 'equal' | 'percentage';
    assigneePercentages?: Record<string, number>; // Map of assignee name to percentage
    assigneePaymentStatus?: Record<string, string>; // Map of assignee name to payment status ('paid' | 'unpaid')
}

export interface Receipt {
    uid?: string;
    vendor?: string;
    receiptDate?: string;
    currency?: string;
    subtotal?: number;
    tax?: number;
    taxPercentage?: number;
    tip?: number;
    tipPercentage?: number;
    total?: number;
    lineItems?: LineItem[];
    userId?: string;
    createdAt?: number; // Epoch milliseconds
    expiresAt?: number; // Epoch milliseconds
    paymentStatus?: 'pending' | 'partial' | 'complete';
    assigneePaymentStatus?: Record<string, string>; // Map of assignee name to payment status ('paid' | 'unpaid')
    taxTipDistribution?: 'proportional' | 'even'; // How tax and tip are distributed
}

export interface ErrorResponse {
    error: string;
}

export interface SaveReceiptResponse {
    saved: boolean;
}
